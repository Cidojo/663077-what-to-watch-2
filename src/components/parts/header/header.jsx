import * as React from 'react';
import * as PropTypes from 'prop-types';
import {Logo} from './../logo/logo.jsx';
import {UserBlock} from './../user-block/user-block.jsx';

const Header = (props) => {
  const {extraClassName, title, isWithUserBlock} = props;
  return (
    <header className={`page-header ${extraClassName}`}>
      <Logo/>

      {title && <h1 className="page-title user-page__title">{title}</h1>}

      {isWithUserBlock && <UserBlock/>}
    </header>
  );
};

Header.propTypes = {
  extraClassName: PropTypes.string,
  title: PropTypes.string,
  isWithUserBlock: PropTypes.bool
};

Header.defaultProps = {
  extraClassName: `movie-card__head`,
  title: ``,
  isWithUserBlock: false
};

export {Header};
