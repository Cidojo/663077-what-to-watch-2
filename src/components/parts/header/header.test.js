import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Header} from './header.jsx';

configure({adapter: new Adapter()});

const props = {
  extraClassName: ``,
  title: ``,
  isWithUserBlock: false
};

it(`should render Header component without errors`, () => {
  const tree = shallow(<Header {...props}/>);
  expect(toJson(tree)).toMatchSnapshot();
});

it(`should render Header component without errors when with UserBlock`, () => {
  const tree = shallow(<Header {...props} isWithUserBlock={true} />);
  expect(toJson(tree)).toMatchSnapshot();
});
