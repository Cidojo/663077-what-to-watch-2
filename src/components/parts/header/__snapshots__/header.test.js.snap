// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component without errors 1`] = `
<header
  className="page-header "
>
  <Logo />
</header>
`;

exports[`should render Header component without errors when with UserBlock 1`] = `
<header
  className="page-header "
>
  <Logo />
  <Connect(UserBlock) />
</header>
`;
